### PR TITLE
Increase coverage for pkg/virt-handler

### PR DESCRIPTION
**Pull request by dhiller**
**What this PR does / why we need it**:
In an effort to counter the ever decreasing coverage we add some code path execution tests for `pkg/virt-handler`.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/virt-handler/vm_test.go [hunk 1]
```diff
@@ -3254,6 +3318,58 @@ var _ = Describe("DomainNotifyServerRestarts", func() {
...
+			},
+		}, "/var/run/kubevirt-private/secret-disks/test.iso", true),
+		table.Entry("serviceAccount", &v1.Volume{
+			VolumeSource: v1.VolumeSource{
+				ServiceAccount: &v1.ServiceAccountVolumeSource{},
+			},
+		}, "/var/run/kubevirt-private/service-account-disk/service-account.iso", true),
+		table.Entry("sysprep", &v1.Volume{
+			Name: "test",
+			VolumeSource: v1.VolumeSource{
```
Comment (rmohr) [inline pkg/virt-handler/vm_test.go:3363]: You could reduce the scope even to the `VolumeSource` as parameter and construct the VM in the test to make it even more focused.

File: pkg/virt-handler/vm.go [hunk 2]
```diff
@@ -1144,24 +1143,31 @@ func (d *VirtualMachineController) updateFSFreezeStatus(vmi *v1.VirtualMachineIn
 
 }
 
+const (
+	basepath                  = "/var/run"
+	ephemeralDisksPathSegment = "kubevirt-ephemeral-disks"
+	cloudInitPathSegment      = "cloud-init-data"
+	privatePathSegment        = "kubevirt-private"
+	noCloudIsoFileName        = "noCloud.iso"
+)
```
Comment (iholder-redhat) [inline pkg/virt-handler/vm.go:1152]: aweomse :+1:

File: pkg/virt-handler/vm_test.go [hunk 3]
```diff
@@ -1862,21 +1864,37 @@ var _ = Describe("VirtualMachineInstance", func() {
...
+						CloudInitNoCloud: &v1.CloudInitNoCloudSource{},
+					},
+				})
+				vmi.Status.VolumeStatus = append(vmi.Status.VolumeStatus, v1.VolumeStatus{Name: volumeName})
+				vmi.Status.Phase = v1.Running
+				vmi.Status.NodeName = host
+				vmi = addActivePods(vmi, podTestUUID, host)
+
+				noCloudISOPath := filepath.Join(mockIsolationResult.MountRoot(), basepath, ephemeralDisksPathSegment, cloudInitPathSegment, vmi.Namespace, vmi.Name)
+				err = os.MkdirAll(noCloudISOPath, 0755)
```
Comment (iholder-redhat) [inline pkg/virt-handler/vm_test.go:1887]: Is this being eventually deleted?

File: pkg/virt-handler/vm_test.go [hunk 4]
```diff
@@ -1833,6 +1835,68 @@ var _ = Describe("VirtualMachineInstance", func() {
...
+			domain := api.NewMinimalDomainWithUUID("testvmi", vmiTestUUID)
+			domain.Status.OSInfo = api.GuestOSInfo{
+				Name: "new-name",
+			}
+			vmi := api2.NewMinimalVMI("testvmi")
+			vmi.Status.GuestOSInfo = v1.VirtualMachineInstanceGuestOSInfo{
+				Name: "old-name",
+			}
+			controller.updateGuestInfoFromDomain(vmi, domain)
+		})
```
Comment (iholder-redhat) [inline pkg/virt-handler/vm_test.go:1865]: These tests do not actually test anything. They don't expect any values nor validate order of execution or anything like that.
Am I missing anything, or is it just pure-raising-coverage code?

IMHO - the bare minimum here is to at least test the values that these functions return.

Comment (iholder-redhat) [reply pkg/virt-handler/vm_test.go:1865]: (As a side note - see that all the error checking within these functions are not being covered - which is totally fine by me :) )

**Review by iholder-redhat — COMMENTED** [review]
Thank you @dhiller!

File: pkg/virt-handler/vm_test.go (hunk 4 above)
Comment (dhiller) [reply pkg/virt-handler/vm_test.go:1865]: Hey, thanks for the review!

Actually, these are code path tests for the "happy path", so I didn't even bother checking the return values. At least those make sure the code doesn't fail :) . But yes, I can add those verifications.

**Pull request closed without merging**